coordinator/src/node.rs

```diff
@@ -573,10 +573,30 @@ impl Node {
             self.finalize_rollover(&r.channel_id)?;
         }
 
+        if let Message::SubChannel(SubChannelMessage::Finalize(finalize)) = &msg {
+            tracing::warn!("Subchannel offer was finalized {}", finalize.channel_id);
+            let mut connection = self.pool.get()?;
+            db::positions::Position::update_proposed_position(
+                &mut connection,
+                node_id.to_string(),
+                PositionState::Open,
+            )?;
+        }
+
         if let Message::SubChannel(SubChannelMessage::CloseFinalize(_msg)) = &msg {
             self.continue_position_resizing(node_id)?;
         }
 
+        if let Message::SubChannel(SubChannelMessage::Reject(reject)) = &msg {
+            tracing::warn!("Subchannel offer was rejected {}", reject.channel_id);
+            let mut connection = self.pool.get()?;
+            db::positions::Position::update_proposed_position(
+                &mut connection,
+                node_id.to_string(),
+                PositionState::Failed,
```

Comment:
🤔 i am not sure about the failed state of a position. The order can fail, but the position? IMHO the position should be in the state is has been before the order.

E.g. if the position is `Open` and the `CloseOffer` gets rejected, the position should remain in `Open` and not in `Failed`. 

But the order should go to failed. Btw. The matched order will also have to be handled in that case.

Reply:
Can I get the explanation too, @bonomat? You probably convinced @holzeis offline.

My guess is that this only applies to channels _without_ a position, so transitioning from `Proposed` to `Failed` will not overwrite an `Open` position.

And what about the order, like Richie mentioned?

Reply:
> My guess is that this only applies to channels without a position, so transitioning from Proposed to Failed will not overwrite an Open position.

This is correct. We only update the state from `proposed` positions here because a user might have some passed `closed` positions which we dont' want to update. 

> 🤔 i am not sure about the failed state of a position. The order can fail, but the position? IMHO the position should be in the state is has been before the order.

Thanks for pointing this out. If the position has been `proposed`, i.e. it basically does not exist, it should go into `failed` because I don't think there is a way to recover from this - the `proposed` message will always look the same.
However, if this was a `resize` we need to go back into the old state, i.e. `open` as you've pointed out so that we can retry the protocol if needed.

Reply:
> But the order should go to failed. Btw. The matched order will also have to be handled in that case.

How can I identify the order here?

Reply:
> How can I identify the order here?

We should put the match into failed and the order to _un_filled/open. However, at this point I don't know which order/match it was. Given these are actually two different systems we need to rethink this concept. 
I would opt for not dealing with this here and accepting that we might have some matches in an invalid state because the subchannel was never opened. 

Same goes for `Trades`: this should also be put into `failed` if the subchannel protocol failed.